sdk/core/azure_core_test/src/lib.rs

```diff
@@ -7,11 +7,14 @@ pub mod proxy;
 pub mod recorded;
 mod recording;
 
-pub use azure_core::test::TestMode;
+use azure_core::Error;
+pub use azure_core::{error::ErrorKind, test::TestMode};
 pub use proxy::{matchers::*, sanitizers::*};
 pub use recording::*;
 use std::path::{Path, PathBuf};
 
+#[cfg_attr(target_arch = "wasm32", allow(dead_code))]
```

Comment:
Why do we want to allow dead_code? Add a comment

Reply:
This tends to be common in libs, actually. Just about everything you depend on has this either targeted (as we do and should) or general e.g., `#![allow(dead_code)]` which seems lazy. It's because, as a library, we can't always test everything and we expose some public APIs that don't otherwise get touched (any `Debug`-derived printing is ignored by the linter).